src/cluster.c

```diff
@@ -44,7 +44,8 @@ char *ShardGroupSerialize(ShardGroup *sg)
     for (unsigned int i = 0; i < sg->slot_ranges_num; i++) {
         /* individual slot ranges */
         ShardGroupSlotRange *sr = &sg->slot_ranges[i];
-        buf = catsnprintf(buf, &buf_size, "%u\n%u\n%u\n", sr->start_slot, sr->end_slot, sr->type);
+        /* storing magic as an unsigned long long as there's a strtoull() function, not a stroll() */
+        buf = catsnprintf(buf, &buf_size, "%u\n%u\n%u\n%llu\n", sr->start_slot, sr->end_slot, sr->type, sr->magic);
```

Comment:
actually there is `strtoll` but, in any case `magic` probably should be an `unsigned`, so let's change it in `ShardGroupSlotRange` definition. Beside that - I think we should change `magic` to some more meaningful name - to indicate its real meaning and values.